Assets/MRTK/SDK/Experimental/Dock/Dockable.cs

```diff
@@ -0,0 +1,335 @@
+            {
+                Assert.IsNotNull(dockedPosition, "When a dockable is docked, its dockedPosition must be valid.");
+                Assert.AreEqual(dockedPosition.dockedObject, this, "When a dockable is docked, its dockedPosition reference the dockable.");
+
+                var lerpTime = dockingState == DockingState.Docked ? moveLerpTimeWhenDocked : moveLerpTime;
+
+                if (!isDragging)
+                {
+                    // Don't override dragging
+                    transform.position = Solver.SmoothTo(transform.position, dockedPosition.transform.position, Time.deltaTime, lerpTime);
+                    transform.rotation = Solver.SmoothTo(transform.rotation, dockedPosition.transform.rotation, Time.deltaTime, lerpTime);
+                }
+
+                transform.localScale = Solver.SmoothTo(transform.localScale, dockedPositionScale, Time.deltaTime, lerpTime);
+
+                if (CloseEnough(dockedPosition.transform.position, transform.position) &&
+                    AlignedEnough(dockedPosition.transform.rotation, transform.rotation) &&
+                    AboutTheSameSize(dockedPositionScale, transform.localScale))
+                {
+                    // Finished docking
+                    dockingState = DockingState.Docked;
+
+                    // Snap to position
+                    transform.position = dockedPosition.transform.position;
+                    transform.rotation = dockedPosition.transform.rotation;
+                    transform.localScale = dockedPositionScale;
+                }
+            }
+            else if (dockedPosition == null && dockingState == DockingState.Undocking)
+            {
+                transform.localScale = Solver.SmoothTo(transform.localScale, originalScale, Time.deltaTime, moveLerpTime);
+
+                if (AboutTheSameSize(originalScale, transform.localScale))
+                {
+                    // Finished undocking
+                    dockingState = DockingState.Undocked;
+
+                    // Snap to size
+                    transform.localScale = originalScale;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Docks this object in a given <see cref="DockPosition"/>.
+        /// </summary>
+        /// <param name="position">The <see cref="DockPosition"/> where we'd like to dock this object.</param>
+        public void Dock(DockPosition position)
+        {
+            if (!CanDock)
+            {
+                Debug.LogError($"Trying to dock an object that was not undocked. State = {dockingState}");
+                return;
+            }
+
+            Debug.Log($"Docking object {gameObject.name} on position {position.gameObject.name}");
+
+            dockedPosition = position;
+            dockedPosition.dockedObject = this;
+            float scaleToFit = ScaleToFit(gameObject.GetComponent<Collider>().bounds, dockedPosition.GetComponent<Collider>().bounds);
```

Comment:
check out 'BoundsExtensions.GetScaleToFit()' - this could be an overload.

Reply:
Added a new method BoundsExtensions.GetScaleToFitInside, the current one returns a Vector3 and not a single scale factor

Reply:
Works for me - though you can overload with a different return type.

Reply:
I added a new method because the name didn't match either (BoundsExtensions.GetScaleToMatch())